api/src/main/java/org/pmiops/workbench/cohortbuilder/FieldSetQueryBuilder.java

```diff
@@ -202,69 +405,80 @@ private void handleResultFilters(ResultFilters resultFilters,
         operator = "or";
         childFilters = resultFilters.getAnyOf();
       }
-      sqlBuilder.append("(");
+      queryState.whereSql.append("(");
       boolean first = true;
       for (ResultFilters childFilter : childFilters) {
         if (first) {
           first = false;
         } else {
-          sqlBuilder.append("\n");
-          sqlBuilder.append(operator);
-          sqlBuilder.append("\n");
+          queryState.whereSql.append("\n");
+          queryState.whereSql.append(operator);
+          queryState.whereSql.append("\n");
         }
-        handleResultFilters(childFilter, tableQueryAndConfig, sqlBuilder, paramMap);
+        handleResultFilters(childFilter, queryState);
+      }
+      queryState.whereSql.append(")\n");
+    }
+  }
+
+  private void handleOrderBy(QueryState queryState) {
+    List<String> orderBy = queryState.tableQuery.getOrderBy();
+    if (orderBy.isEmpty()) {
+      throw new BadRequestException("Order by list must not be empty");
+    }
+    List<String> orderByColumns = new ArrayList<>();
+    for (String columnName : orderBy) {
+      String columnStart = columnName;
+      boolean descending = false;
+      if (columnName.toUpperCase().endsWith(DESCENDING_SUFFIX)) {
+        columnStart = columnName.substring(0, columnName.length() - DESCENDING_SUFFIX.length());
+        descending = true;
       }
-      sqlBuilder.append(")\n");
+      ColumnInfo columnInfo = getColumnInfo(queryState, columnStart);
+      orderByColumns.add(descending ? columnInfo.getColumnName() + " DESC" : columnInfo.getColumnName());
     }
+    queryState.endSql.append("order by ");
+    queryState.endSql.append(Joiner.on(", ").join(orderByColumns));
   }
 
-  public QueryJobConfiguration buildQuery(ParticipantCriteria participantCriteria,
+  public QueryConfiguration buildQuery(ParticipantCriteria participantCriteria,
       TableQueryAndConfig tableQueryAndConfig, long resultSize, long offset) {
-    TableQuery tableQuery = tableQueryAndConfig.getTableQuery();
-    List<String> columnNames = tableQuery.getColumns();
-    String tableName = tableQuery.getTableName();
+    QueryState queryState = new QueryState();
```

Comment:
I took all of the SQL builders except fromSql out of QueryState.

fromSql gets updated whenever we need to join in another table, which can happen during processing of any part of the query. Eventually we may want to change the code to parse stuff into an intermediate representation as a first step, or build up some data structure instead of SQL that we later turn into a bunch of joins. For now hopefully this is OK.

I replaced tableQuery with mainTableName. I like being able to use the main table's name for error messages everywhere... wrapped exceptions can be kinda ugly to read. But you're right that we don't need the whole query everywhere.

schemaConfig does get used everywhere we need to do a join, which is throughout the code, so I left that in.